internal/eval/scatter.go

```diff
@@ -0,0 +1,30 @@
+package eval
+
+// Scatter file is used by shardThread to evaluate
+// individual commands on each shard. Scatter functions
+// returns response as interface and error using
+// a structure for each command, which eventiually passed
+// till worker from each shard response
+
+import (
+	"fmt"
+)
+
+type EvalScatterResponse struct {
+	Result interface{} // Result of the Store operation, for now the type is set to []byte, but this can change in the future.
+	Error  error
+}
+
+// ScatterPING function is used by shardThread to
+// evaluate PING command
+func ScatterPING(args []string) EvalScatterResponse {
```

Comment:
IMO scatter should be used convey dividing and sending the requests to store/shard and gather should be used to convey receiving and consolidating the store/shard, and both of them seem to be functions of the worker.

Reply:
I would think that we can later add Gather at command level, if the reduce logic/gather logic is complicated. For the most part, it should be a generic logic of appending all result sets.

Reply:
Scatter shouldn't exist at command level.